allennlp/commands/evaluate.py

```diff
@@ -0,0 +1,91 @@
+from typing import Dict, Any
+import argparse
+import json
+import os
+import logging
+
+from allennlp.common.params import Params, replace_none
+from allennlp.data import Vocabulary, Dataset
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.iterators import DataIterator
+from allennlp.models.model import Model
+from allennlp.nn.util import arrays_to_variables, device_mapping
+
+import torch
+import tqdm
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+def add_subparser(parser: argparse._SubParsersAction) -> argparse.ArgumentParser:  # pylint: disable=protected-access
+    description = '''Evaluate the specified model + dataset'''
+    subparser = parser.add_parser(
+            'evaluate', description=description, help='Evaluate the specified model + dataset')
+    subparser.add_argument('--config_file',
+                           type=str,
+                           required=True,
+                           help='path to the configuration file that trained the model')
+    subparser.add_argument('--weights_file',
```

Comment:
If you have to provide something outside of the experiment config anyway, I think being explicit about it and just providing the path is very reasonable. It also allows you to use this function if you used mainly allennlp components, but a different training procedure.